src/SourceBuild/content/.devcontainer/prebuilt-sdk/init.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+source="${BASH_SOURCE[0]}"
+script_root="$( cd -P "$( dirname "$source" )" && pwd )"
+
+"$script_root"/../../prep.sh
+
+cp "$script_root/../synchronize-vmr.sh" "$script_root/../../.."
+
+# GitHub Codespaces sets this and it conflicts with source-build scripts.
+unset RepositoryName
```

Comment:
We should do something with this unset directive. At least understanding how could this interfere with the source-build scripts would be good.

Reply:
I agree. @MichaelSimons would you know if we can put this in the `build.sh` or it would break some other flow somewhere which sets this to something and gets some other behaviour? I wasn't able to find much about this env.

Reply:
Okay, I went through the code and I think it's safe to unset this variable in the top-level `build.sh` in VMR for a number of reasons:
- This script is not called again recursively so it won't influence anything later.
- Setting this to any repo's name fails the build.
- Setting this to any other string fails the build.

This doesn't work because the variable is read from env here: https://github.com/dotnet/dotnet/blob/e60e531e91328cc30e0559ae7ace03b84718a38c/repo-projects/Directory.Build.props#L4

And it's expected that it will match the name of the currently built repo/project.

Reply:
The source build [RepositoryName property](https://github.com/dotnet/dotnet/blob/e60e531e91328cc30e0559ae7ace03b84718a38c/repo-projects/Directory.Build.props#L4) extension mechanism does not seem to be necessary.  Instead of clearing it in the build script, I propose we remove the extension point.  Let's not add more logic around something not being used.

Reply:
@MichaelSimons do you mean "remove the condition"?

Reply:
> @MichaelSimons do you mean "remove the condition"?

Yes.

Reply:
Done

Reply:
@MichaelSimons this is too an invasive change since it's in `Directory.Build.props` and breaks the build. `unset` in `build.sh` works fine.

Reply:
Can you provide an error?  I am curious how this breaks the build.

Reply:
The reason this broke appears because you didn't remove the reference you noted earlier in source-build-externals.  I think that can also be removed.

Reply:
I did remove that but it wasn't a clean working tree. I will bump sb externals in this PR and give it another go

Reply:
Ah, a fresh build seemed to work fine